### PR TITLE
register API types only once for encryption config

### PR DESCRIPTION
**Pull request by aramase**
/kind cleanup

```release-note
NONE
```

/sig auth
/triage accepted
/milestone v1.29

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go [hunk 1]
```diff
@@ -104,6 +104,16 @@ const (
...
 
+var (
+	configScheme = runtime.NewScheme()
+	codecs       = serializer.NewCodecFactory(configScheme)
+)
+
+func init() {
+	utilruntime.Must(apiserverconfig.AddToScheme(configScheme))
+	utilruntime.Must(apiserverconfigv1.AddToScheme(configScheme))
+}
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go:115]: I think we only need the `codecs` global?

```suggestion
var codecs serializer.CodecFactory

func init() {
	configScheme := runtime.NewScheme()
	utilruntime.Must(apiserverconfig.AddToScheme(configScheme))
	utilruntime.Must(apiserverconfigv1.AddToScheme(configScheme))
	codecs = serializer.NewCodecFactory(configScheme)
}
```

Comment (aramase) [reply staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go:115]: that's correct!

**Pull request merged**